### PR TITLE
Add Quickstart for SolidJS

### PR DESCRIPTION
**Pull request by johnhenry**
## What kind of change does this PR introduce?

Docs Update: Add a Quickstart for [SolidJS](https://www.solidjs.com/)
https://github.com/supabase/supabase/issues/6388

### CODE REVIEW

File: web/docs/guides/with-solidjs.mdx [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+```bash
+npm install @supabase/supabase-js
+```
+
+And finally we want to save the environment variables in a `.env`.
+All we need are the API URL and the `anon` key that you copied [earlier](#get-the-api-keys).
+
+```bash title=".env"
+VITE_SUPABASE_URL=YOUR_SUPABASE_URL
+VITE_SUPABASE_ANON_KEY=YOUR_SUPABASE_ANON_KEY
```
Comment (kiwicopple) [inline web/docs/guides/with-solidjs.mdx:185]: Any reason why these are prefixed with `VITE`?

File: web/docs/guides/with-solidjs.mdx [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+}
+```
+
+### Account page
+
+After a user is signed in we can allow them to edit their profile details and manage their account.
+
+Let's create a new component for that called `Account.js`.
+
+```jsx title="src/Account.jsx"
```
Comment (kiwicopple) [inline web/docs/guides/with-solidjs.mdx:265]:
```suggestion
```jsx title="src/Account.js"
```

File: web/docs/guides/with-solidjs.mdx [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+```
+
+And one optional step is to update the CSS file `src/index.css` to make the app look nice.
+You can find the full contents of this file [here](https://raw.githubusercontent.com/supabase/supabase/master/examples/react-user-management/src/index.css).
+
+### Set up a Login component
+
+Let's set up a SolidJS component to manage logins and sign ups. We'll use Magic Links, so users can sign in with their email without using passwords.
+
+```jsx title="/src/Auth.jsx"
```
Comment (kiwicopple) [inline web/docs/guides/with-solidjs.mdx:207]:
```suggestion
```jsx title="/src/Auth.js"
```

File: web/docs/guides/with-solidjs.mdx [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+export default Account
+```
+
+### Launch!
+
+Now that we have all the components in place, let's update `App.js`:
+
+```jsx title="src/App.jsx"
```
Comment (kiwicopple) [inline web/docs/guides/with-solidjs.mdx:382]:
```suggestion
```jsx title="src/App.js"
```

File: web/docs/guides/with-solidjs.mdx [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+    </div>
+  )
+}
+```
+
+### Add the new widget
+
+And then we can add the widget to the Account page:
+
+```jsx title="src/Account.jsx"
```
Comment (kiwicopple) [inline web/docs/guides/with-solidjs.mdx:514]:
```suggestion
```jsx title="src/Account.js"
```

**Review by kiwicopple — APPROVED** [review]
this is amazing. Thank you @johnhenry 

A few small suggestions. Looking forward to merging this

File: web/docs/guides/with-solidjs.mdx (hunk 2 above)
Comment (johnhenry) [reply web/docs/guides/with-solidjs.mdx:265]: My initial goal was to make this similar to the React Quickstart with as few changes as possible. In fact, 
I initially tried getting this to work with the ".js" extension, but the tooling expects ".jsx".

**Review by kiwicopple — APPROVED** [review]
Thanks for this @johnhenry - it is an awesome contribution. We'd love to send you some swag. 

If you want a tshirt, could you please send you shirt size and postal address to swag@supabase.io? (also include a link to this PR)

**Pull request merged**